### PR TITLE
Merge dev into book pages

### PR DESCRIPTION
**Pull request by OptimumOpium**

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -166,58 +168,69 @@ tasks.register('integrationTest', Test) {
 }
 
 dependencies {
-    api platform('com.exactpro.th2:bom:3.2.0')
-    api "com.exactpro.th2:cradle-core:${cradleVersion}"
+    api platform("com.exactpro.th2:bom:4.0.2")
+    api ("com.exactpro.th2:cradle-core:${cradleVersion}") {
+        exclude group: 'org.slf4j', module: 'slf4j-log4j12' // because of the vulnerability
```
Comment (Nikita-Smirnov-Exactpro) [inline build.gradle:173]: Does cradle-core:5 contain slf3j-log4j12 ?

File: build.gradle [hunk 2]
```diff
@@ -166,58 +168,69 @@ tasks.register('integrationTest', Test) {
...
+    }
     api 'com.exactpro.th2:grpc-common:4.0.0-th2-2150-books-pages-2635598705-SNAPSHOT'
 
     implementation 'com.google.protobuf:protobuf-java-util'
     implementation 'com.exactpro.th2:grpc-service-generator:3.2.2'
-    implementation "com.exactpro.th2:cradle-cassandra:${cradleVersion}"
+    implementation ("com.exactpro.th2:cradle-cassandra:${cradleVersion}") {
+        exclude group: 'org.slf4j', module: 'slf4j-log4j12' // because of the vulnerability
```
Comment (Nikita-Smirnov-Exactpro) [inline build.gradle:180]: The same question as cradle-core

File: src/main/java/com/exactpro/th2/common/schema/message/impl/rabbitmq/AbstractRabbitSubscriber.java [hunk 3]
```diff
@@ -128,6 +128,8 @@ public void start() throws Exception {
                                         try {
                                             value = valueFromBytes(delivery.getBody());
                                         } catch (Exception e) {
+                                            LOGGER.error("Couldn't parse delivery. Confirm message received", e);
+                                            confirmation.confirm();
```
Comment (Nikita-Smirnov-Exactpro) [inline src/main/java/com/exactpro/th2/common/schema/message/impl/rabbitmq/AbstractRabbitSubscriber.java:132]:
```suggestion
                                            confirmation.reject();
```

File: src/main/java/com/exactpro/th2/common/schema/message/impl/rabbitmq/AbstractRabbitSubscriber.java [hunk 4]
```diff
@@ -236,6 +238,11 @@ protected void handle(String consumeTag, Delivery delivery, T value, Confirmatio
             }
         } catch (Exception e) {
             LOGGER.error("Can not parse value from delivery for: {}", consumeTag, e);
+            try {
+                confirmation.confirm();
```
Comment (Nikita-Smirnov-Exactpro) [inline src/main/java/com/exactpro/th2/common/schema/message/impl/rabbitmq/AbstractRabbitSubscriber.java:242]:
```suggestion
                confirmation.reject();
```

**Review by Nikita-Smirnov-Exactpro — CHANGES_REQUESTED** [review]

**Comment by OptimumOpium** [comment]
Closed because the newer branch for cradle 5.0 exists

**Pull request closed without merging**